sdk/python/pulumi_kubernetes/apiextensions/CustomResource.py

```diff
@@ -15,37 +19,67 @@ class CustomResource(pulumi.CustomResource):
     `ServiceMonitor` resource definition as an argument.
     """
 
-    def __init__(self, __name__, __opts__=None, api_version=None, kind=None, metadata=None, spec=None):
-        if not __name__:
+    def __init__(
+            self,
+            resource_name: str,
+            api_version: str,
+            kind: str,
+            spec: pulumi.Input[Any] = None,
```

Comment:
on the fence here.  we're diverging from our terraform generation in terms of adding these annotations to the params.  that said, the annotations are good...   

@lukehoban any opinions on this space?  We could:

1. try to keep k-sdk consistent with existing terraform sdks.
2. do this, and move the terraform sdks to this.
3. not care and allow them to diverge.

Reply:
Note that we also generate propere doc comments for the `__init__ function in the TF SDKs.

https://github.com/pulumi/pulumi-aws/blob/474395e64fb9414b36f96006970f96d0b93ac885/sdk/python/bin/build/lib/pulumi_aws/alb/listener_rule.py#L33-L46

I do think being consistent is valuable here.  In theory all of these annotations (types on params and doc comments) are goodness, but I do think we should aim for consistency first, as that will ensure docs and IDE experience are at least consistent.

Reply:
@lukehoban I'm not clear from your comment if we should keep the type annotations or not?

Reply:
> but I do think we should aim for consistency first, as that will ensure docs and IDE experience are at least consistent.

I think this means  keep kubernetes consistent with terraform.  That would be done with two changes:

1. remove the direct type-annotations from the code.  So  `spec = None` instead of ` spec: pulumi.Input[Any] = None`.  
2. Add an appropriate doc comment to the function that is very close to how we doc things in the terraform generated code.  i.e., like this: https://github.com/pulumi/pulumi-aws/blob/d42f7d16700e2cd9b68264cd4135b7de6f37af25/sdk/python/pulumi_aws/lb/listener_rule.py#L33-L43

Reply:
In general, we should go for consistency.  Then, if we think there's something we can do better with teh codegen, we can do it for both kubernetes/terraform.  Every time we deviate, it's just something that is difficult ot deal with as we need to evolve/fix things.  

Thanks!